core/src/test/java/org/elasticsearch/bwcompat/OldIndexBackwardsCompatibilityIT.java

```diff
@@ -176,7 +176,8 @@ void unloadIndex(String indexName) throws Exception {
     public void testAllVersionsTested() throws Exception {
         SortedSet<String> expectedVersions = new TreeSet<>();
         for (Version v : VersionUtils.allReleasedVersions()) {
-            if (VersionUtils.isSnapshot(v)) continue;  // snapshots are unreleased, so there is no backcompat yet
+            // The current version is in the "released" list even though it isn't released for historical reasons
```

Comment:
what do you mean "historical reasons"? This comment is confusing without further explanation.